### PR TITLE
Public journals update

### PR DESCRIPTION
**Pull request by kompotkot**
## Changes

Public journal changes:
- fixed exceptions handling
- extended cli commands
- updated list of public journals and public hournal check endpoints
- create new entry possibility as unregistered user

## How to test these changes?

Tested locally with entity

### CODE REVIEW

**Comment by kompotkot** [comment]
@bugout-dev check
- [ ] Remove BLOF journal from public access (use CLI after deployment)
- [ ] Merge bugout frontend update with public user id for welcome journal

File: spire/public/api.py [hunk 1]
```diff
@@ -76,62 +72,75 @@ async def version() -> VersionResponse:
...
+    user_id: UUID = Query(...),
+    db_session: Session = Depends(yield_db_read_only_session),
+) -> BugoutJournals:
+    """
+    List journals with public access.
+    """
+    try:
+        public_user = actions.get_public_user(db_session=db_session, user_id=user_id)
+        result: BugoutJournals = bugout_api.list_journals(
+            token=public_user.restricted_token_id
```
Comment (zomglings) [inline spire/public/api.py:86]: Should we restrict this to list only journals whose journal id is in `public_journals.journal_id`?

File: spire/public/api.py [hunk 2]
```diff
@@ -143,22 +152,117 @@ async def get_public_journal_handler(
...
+                    detail=f"Field {f['name']} length too long for public access",
+                )
+
+    for tag in create_request.tags:
+        if len(tag) > 50:
+            raise HTTPException(
+                status_code=413, detail="Tag length too long for public access"
+            )
+    try:
+        public_user = actions.get_public_journal_user(
```
Comment (zomglings) [inline spire/public/api.py:207]: @kompotkot : Can anyone create entries in any public journal? What if we remove `journal.entries.create` permission from public user on a public journal?

Maybe we should do permission check? And maybe we should take argument of permissions for public user when we create that user for a public journal?

**Review by zomglings — COMMENTED** [review]
Few questions @kompotkot

**Comment by zomglings** [comment]
@kompotkot : Did you update deployment env vars?

File: spire/public/api.py (hunk 1 above)
Comment (kompotkot) [reply spire/public/api.py:86]: I do not think so, if this user (we operate as autogenerated public user) has access to journal, then someone granted permissions to read a journal.

File: spire/public/api.py (hunk 2 above)
Comment (kompotkot) [reply spire/public/api.py:207]: Yes, anyone can create entries in public journal if such permissions granted to public user with cli. If we will remove `journal.entries.create` permission, it means that user became just a read-only user. There is no consistency in database which we need to check, we are duplicate what permissions we granted to user.

I think we can add endpoint to check what kind of permissions do we have according to some journal.

**Review by Andrei-Dolgolev — APPROVED** [review]
Looks correct

**Pull request merged**